dev-tools/mage/gotool/go.go

```diff
@@ -37,6 +37,19 @@ type Args struct {
 // ArgOpt is a functional option adding info to Args once executed.
 type ArgOpt func(args *Args)
 
+type goInstall func(opts ...ArgOpt) error
```

Comment:
I see, we copied gotool from `go-txfile` to the Beats repo. Btw. `go-txfile` [magefile](https://github.com/elastic/go-txfile/blob/master/magefile.go) uses the gotools package (minor improvements + generalization of this approach) from `https://github.com/urso/magetools`.